### PR TITLE
flake8 format checking in ci build

### PR DESCRIPTION
**Pull request by iblis17**
Let's dicusse our formating criteria here. And doing the format checking in travis.

Ref: https://flake8.readthedocs.org/en/latest/config.html

I will propse that
```ini
max-length = 100
```

Please comment.

### CODE REVIEW

**Comment by iblis17** [comment]
Ok, we got lot of tragedies reported by `flake8`.

Do we need to fix them in this PR or later?

**Comment by M157q** [comment]
Let's fix them all in this PR!

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,9 @@
 [pytest]
-addopts = -v --cov zdict --cov-report html --cov-report term
+flake8-ignore =
+    zdict/__init__.py ALL
+    zdict/dictionaries/template.py ALL
+
+addopts = --flake8 -v --cov zdict --cov-report html --cov-report term
```
Comment (iblis17) [inline setup.cfg]: Do not add `--flake8` option here.
I have made it in travis ci build matrix.

File: setup.cfg [hunk 2]
```diff
@@ -1,5 +1,9 @@
 [pytest]
-addopts = -v --cov zdict --cov-report html --cov-report term
+flake8-ignore =
+    zdict/__init__.py ALL
```
Comment (iblis17) [inline setup.cfg]: I think ignoring `F401` only will be better.
```
zdict/__init__.py F401
```

File: setup.cfg (hunk 1 above)
Comment (M157q) [inline setup.cfg]: It's for using py.test in local env.

File: setup.cfg (hunk 2 above)
Comment (iblis17) [inline setup.cfg]: And maybe we should take care other `__init__.py`.
Can we use glob to match pattern of `__init__.py`?

File: setup.cfg (hunk 1 above)
Comment (iblis17) [inline setup.cfg]: Accutually, travis will be affected by `setup.cfg`.

We invoke `setup.py test` in travis, so this config will pollute travis.

Comment (M157q) [inline setup.cfg]: I don't get it.
Aren't we going to use the flake8 in Travis?

Comment (iblis17) [inline setup.cfg]: I setup a different build for style checking. Please checkout [travis ci env var](https://travis-ci.org/zdict/zdict/builds/120416912).

I want the other build for unit testing.
Only one build for style checking.

File: setup.cfg (hunk 2 above)
Comment (M157q) [inline setup.cfg]: We don't have any content in other `__init__.py` files currently.
Besides, I don't think we will add things into new `__init__.py` files so often.
So, maybe just ignore `zdict/__init__.py` at this time?

File: setup.cfg (hunk 1 above)
Comment (iblis17) [inline setup.cfg]: If you consider that there is no need to add a addictional build matrix, please revert the changes in `.travis.yaml`

Comment (M157q) [inline setup.cfg]: hmm... May I ask what's the purpose for the separation like this?

Comment (M157q) [inline setup.cfg]: ok. so if I want the flake8 syntax checking in local env, I should use `py.test --flake8` instead of `py.test`.
I'm gonna remove the `--flake8` in `setup.cfg`.

Comment (iblis17) [inline setup.cfg]: Originally, i want to distingish the type of build failure, "Is the unittesting failure or my bad style?"
but i just found that the error message of `pytest-flake8` is quite clear to figure out that.
So maybe there is no need for the extra build.
